.github/workflows/e2e.yml

```diff
@@ -19,20 +19,58 @@ on:
         description: 'Alternative Password'
         required: false
         default: ''
+
+permissions: read-all
+
 jobs:
   Run_e2e_tests:
+    name: Run E2E Tests
+    permissions:
+      contents: read
```

Comment:
Since you currently have `permissions: read-all`, then these two lines seem redundant.

Reply:
Hi @GaryJones, `read-all` also has additional `read` permissions like `security-events: read`, `pull-requests: read` etc. This is why the StepSecurity automation adds `contents: read` explicitly here. I am getting similar feedback though on GitHub Action starter workflows as well (see this [issue](https://github.com/actions/starter-workflows/pull/1341)). One proposal is to have a config setting/ checkbox `Only add job level permissions if they are higher than read-all` in the automation, which would be more aligned with what you are looking for. Or would you prefer that the automation not add the workflow level `permissions: read-all` and just add the job level permissions, which are specific?

Reply:
> Or would you prefer that the automation not add the workflow level permissions: read-all and just add the job level permissions, which are specific?

@varunsh-coder OSSF [does not recommend](https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions) that. The reason is that when you add a new job to the workflow, it is easy to forget to add the `permissions:` option to it.

The best way is probably to have the most restrictive `permissions` for the workflow and adjust the per-job permissions as necessary.

PS: that OSSF document is how I have got to know about StepSecurity :-)

Reply:
I found that the default for those additional permissions is (from another repo):

```
Actions: write
Checks: write
Contents: write
Deployments: write
Discussions: write
Issues: write
Metadata: read
Packages: write
Pages: write
PullRequests: write
RepositoryProjects: write
SecurityEvents: write
Statuses: write
```

So, in that sense, adding `permission: read-all` makes sense to restrict the `write` permission.

But as noted in [my other comment](https://github.com/Automattic/vip-go-mu-plugins/pull/2815#discussion_r786843895), by using the repo settings, this can be dropped even further down to just:

```
Contents: read
Metadata: read
```

and then only job-level permissions need to be added where absolutely necessary.

For this particular issue, if `permissions:read-all` already includes `content: read`, then it still seems redundant to include `content:read` at the job-level.

Reply:
@GaryJones [`contents: read` is more restrictive than `permissions: read-all`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions)

> Any permission that is absent from the list will be set to none.

That is,

```yaml
permissions:
  contents: read
```

is equivalent to

```yaml
permissions:
  contents: read
  actions: none
  checks: none
  deployments: none
  issues: none
  packages: none
  pull-requests: none
  repository-projects: none
  security-events: none
  statuses: none
```

Reply:
While `permissions: read-all` is equivalent to setting all those permissions above to `read`.

I am trying to set `contents: read` at the global level and see how it works — this seems to be a more restrictive setting.

Reply:
> If you specify the access for any of these scopes, all of those that are not specified are set to none.

Ah - I think that's a key bit of information I was missing. Thank you for that!

Reply:
@varunsh-coder does it make sense to replace the top-level

```yaml
permissions: read-all
```

to

```yaml
permissions:
  contents: read
```
and then adjust the per-job permissions? This seems to be a more restrictive configuration than the original one and it ensures that all additional read or write permissions need to be requested explicitly.

Reply:
Thanks @sjinks for the feedback. Let me discuss this and I will get back.